Compilation issue with TypeScript 5.2.2

### Describe the bug

I'm seeing the following compiler error when using TypeScript 5.2.2:

```
$ npx tsc
src/utils.ts:3:35 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("abitype")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts' or create a local package.json file with `{ "type": "module" }`.

3 import type { AbiParameter } from 'abitype';
                                    ~~~~~~~~~
```

I have these settings in my tsconfig.json:
```
"module": "node16",
"moduleResolution": "node16",
```

I noticed that abitype publishes CommonJS files with a `.js` extension, even though in package.json it says `"type": "module"`. Manually changing `"type": "module"` to `"type": "commonjs"` fixes the compiler error. This seems like an odd fix because ES modules are also published with a `.js` extension, however those are perhaps hardcoded to be interpreted as ES modules because they are reached through the explicit `"import"` entrypoints?

### Link to Minimal Reproducible Example

The playground doesn't work with ES Modules.

### Steps To Reproduce

_No response_

### Package Version

0.10.1

I don't see the same issue with 0.9.10.

### TypeScript Version

5.2.2

### Anything else?

_No response_

### Validations

- [X] Checked there isn't [already an issue](https://github.com/wagmi-dev/abitype/issues) that exists for the bug you encountered.
- [X] Followed the [Code of Conduct](https://github.com/wagmi-dev/.github/blob/main/CODE_OF_CONDUCT.md).

--- a/.changeset/smart-students-talk.md
+++ b/.changeset/smart-students-talk.md
@@ -1,0 +1,5 @@
+---
+"abitype": patch
+---
+
+Fixed error with TypeScript 5.2.2 in CommonJS projects.

--- a/.changeset/smart-students-talk.md
+++ b/.changeset/smart-students-talk.md
@@ -2,4 +2,4 @@
 "abitype": patch
 ---
 
-Fixed error with TypeScript 5.2.2 in CommonJS projects.
+Fixes an error with TypeScript 5.2.2 in CommonJS projects..

--- a/.scripts/formatPackageJson.ts
+++ b/.scripts/formatPackageJson.ts
@@ -30,7 +30,7 @@ for (const packagePath of packagePaths) {
     `${JSON.stringify(packageJson, undefined, 2)}\n`,
   )
 
-  const { devDependencies: _dD, scripts: _s, ...rest } = packageJson
+  const { devDependencies: _dD, scripts: _s, type: _t, ...rest } = packageJson
   await Bun.write(packagePath, `${JSON.stringify(rest, undefined, 2)}\n`)
 }
 

--- a/packages/abitype/package.json
+++ b/packages/abitype/package.json
@@ -7,7 +7,7 @@
   "scripts": {
     "build": "pnpm run clean && pnpm run build:cjs && pnpm run build:esm+types",
     "build:cjs": "tsc --project tsconfig.build.json --module commonjs --moduleResolution node10 --outDir ./dist/cjs --removeComments --verbatimModuleSyntax false && echo > ./dist/cjs/package.json '{\"type\":\"commonjs\"}'",
-    "build:esm+types": "tsc --project tsconfig.build.json --module es2020 --outDir ./dist/esm --declaration --declarationMap --declarationDir ./dist/types",
+    "build:esm+types": "tsc --project tsconfig.build.json --module es2020 --outDir ./dist/esm --declaration --declarationMap --declarationDir ./dist/types && echo > ./dist/esm/package.json '{\"type\":\"module\",\"sideEffects\":false}'",
     "clean": "rm -rf dist tsconfig.tsbuildinfo abis zod",
     "test:build": "publint --strict",
     "typecheck": "tsc --noEmit"

